python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts

```diff
@@ -0,0 +1,57 @@
+import { PromptSchema } from "../../utils/promptUtils";
+
+export const HuggingFaceAutomaticSpeechRecognitionPromptSchema: PromptSchema = {
+  // See https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/pipelines#transformers.AutomaticSpeechRecognitionPipeline
+  // for descriptions.
+
+  // The following supported properties cannot be cleanly modeled in the schema and must be implemented programatically:
+  // tokenizer, feature_extractor, device, decoder
+  input: {
+    type: "object",
+    required: ["data"],
+    properties: {
+      data: {
+        type: "string",
+      },
+      attachments: {
+        type: "array",
+        items: {
+          type: "attachment",
+          required: ["data"],
+          mime_types: ["audio/mpeg", "audio/wav", "audio/webm"],
+          properties: {
+            data: {
+              type: "string",
+            },
+          },
+        },
+      },
+    },
+  },
+  model_settings: {
+    type: "object",
+    properties: {
+      chunk_length_s: {
+        type: "number",
+        description: `The input length for in each chunk. If chunk_length_s = 0 then chunking is disabled (default).`,
+      },
+      stride_length_s: {
+        type: "number",
+        description: `The length of stride on the left and right of each chunk.
+         Used only with chunk_length_s > 0. This enables the model to see more context and 
+         infer letters better than without this context but the pipeline discards the stride 
+         bits at the end to make the final reconstitution as perfect as possible.
+         Defaults to defaults to chunk_length_s / 6`,
+      },
+      framework: {
+        type: "string",
+        enum: ["pt", "tf"],
+        description: `The framework to use, either "pt" for PyTorch or "tf" for TensorFlow. 
+        The specified framework must be installed. If no framework is specified, will 
+        default to the one currently installed. If no framework is specified and both 
+        frameworks are installed, will default to the framework of the model, or to PyTorch if 
+        no model is provided.`,
+      },
+    },
```

Comment:
Let's also include `max_new_tokens`: https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/pipelines#transformers.AutomaticSpeechRecognitionPipeline.__call__

Reply:
They also seem to use the same generation params as from text generation: https://huggingface.co/docs/transformers/en/main_classes/text_generation#transformers.GenerationConfig

I think it would be helpful if we could centralize this and have one schema inherit from another, but not sure how simple that is. Worse case we parse the JSON and do `dict.update()` to the `model_settings` field. if not possible immediately, we can create task for it

Reply:
We can easily just create a common prompt schema and spread it into the object then override what's needed

Reply:
Followup issue to track: https://github.com/lastmile-ai/aiconfig/issues/836